addons/context_actions/CfgContext.hpp

```diff
@@ -250,6 +250,34 @@ class EGVAR(context_menu,actions) {
             statement = QUOTE(_hoveredEntity setUnitLoadout configOf _hoveredEntity);
             icon = "\a3\3den\Data\Displays\Display3DEN\ToolBar\undo_ca.paa";
         };
+        class SwitchWeapon {
+            displayName = "$STR_A3_Switch1";
+            icon = "\a3\ui_f\data\IGUI\Cfg\Actions\reammo_ca.paa";
+            class Primary {
+                displayName = "$STR_A3_RSCDisplayArsenal_Tab_PrimaryWeapon";
+                condition = QUOTE([ARR_2(_hoveredEntity,0)] call FUNC(canSwitchWeapon));
+                statement = QUOTE([ARR_3('zen_common_selectWeapon', [ARR_2(_hoveredEntity, primaryWeapon _hoveredEntity)], _hoveredEntity)] call CBA_fnc_targetEvent);
```

Comment:
Probably move the statement to a function as well - `switchWeapon`.

Reply:
are we concerned with the performance difference between these?
```
['zen_common_selectWeapon', [_unit, _weapon], _unit] call CBA_fnc_targetEvent;
['zen_common_selectWeapon', [_unit, _weaponIndex], _unit] call CBA_fnc_targetEvent;
```
`_weapon` being a string presumably is more costly than `_weaponIndex` a number.